### PR TITLE
Add ability to specify spawn name

### PR DESCRIPTION
**Pull request by tedivm**
Right now the only way to specify a spawn name is by using the web interface, otherwise they are autogenerated. This change adds an option for specifying the name.

### CODE REVIEW

**Comment by edhaase** [comment]
You may want to replace `_.isUndefined` with `name == undefined` (double equals) in case a user passes in null.

File: src/game/rooms.js [hunk 1]
```diff
@@ -968,6 +968,9 @@ exports.make = function(_runtimeData, _intents, _register, _globals) {
         if(!C.CONSTRUCTION_COST[structureType]) {
             return C.ERR_INVALID_ARGS;
         }
+        if(structureType == 'spawn' && typeof name == 'string' && createdSpawnNames.indexOf(name) != -1) {
```
Comment (PostCrafter) [inline src/game/rooms.js]: I think you need to check whether the name is already used by another spawn of spawn construction site here, just like you do below when generating a new name:
```_.any(register.spawns, {name}) ||
                _.any(register.constructionSites, {structureType: 'spawn', name})```

**Comment by artch** [comment]
Please rebase this to the `ptr` branch. It will be included in the next PTR patch.

Also, a corresponding PR to docs repository would be nice.

**Comment by tedivm** [comment]
Rebased- and I'll update the docs as well.

**Pull request merged**